python/lsst/daf/butler/queries/overlaps.py

```diff
@@ -259,10 +277,14 @@ def add_join_operand_connections(self, operand_dimensions: DimensionGroup) -> No
         association between non-overlapping things that we'd want to respect by
         *not* adding a more restrictive automatic join.
         """
-        for a_family, b_family in itertools.pairwise(operand_dimensions.spatial):
-            self._spatial_connections.merge(a_family, b_family)
-        for a_family, b_family in itertools.pairwise(operand_dimensions.temporal):
-            self._temporal_connections.merge(a_family, b_family)
+        for a_family, b_family in itertools.pairwise(getattr(operand_dimensions, kind)):
```

Comment:
Can you pass `operand_dimensions.spatial` or `.temporal` into this method instead of doing `getattr`? It might help `mypy` to figure out types of those?

Reply:
Done.  I also noticed along the way that we've been using `pairwise` here but it really should be `combinations`.